backend/src/main/kotlin/app/ehrenamtskarte/backend/freinet/webservice/FreinetAgencyQueryService.kt

```diff
@@ -0,0 +1,40 @@
+package app.ehrenamtskarte.backend.freinet.webservice
+
+import app.ehrenamtskarte.backend.auth.database.AdministratorEntity
+import app.ehrenamtskarte.backend.auth.service.Authorizer
+import app.ehrenamtskarte.backend.common.webservice.EAK_BAYERN_PROJECT
+import app.ehrenamtskarte.backend.common.webservice.GraphQLContext
+import app.ehrenamtskarte.backend.exception.service.ForbiddenException
+import app.ehrenamtskarte.backend.exception.service.NotEakProjectException
+import app.ehrenamtskarte.backend.exception.service.ProjectNotFoundException
+import app.ehrenamtskarte.backend.exception.service.UnauthorizedException
+import app.ehrenamtskarte.backend.freinet.database.repos.FreinetAgencyRepository
+import app.ehrenamtskarte.backend.freinet.webservice.schema.types.FreinetAgency
+import app.ehrenamtskarte.backend.projects.database.ProjectEntity
+import app.ehrenamtskarte.backend.projects.database.Projects
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import graphql.schema.DataFetchingEnvironment
+import org.jetbrains.exposed.sql.transactions.transaction
+
+@Suppress("unused")
+class FreinetAgencyQueryService {
+    @GraphQLDescription("Returns freinet agency information for a particular region. Works only for the EAK project.")
+    fun getFreinetAgencyByRegionId(dfe: DataFetchingEnvironment, regionId: Int, project: String): FreinetAgency? = transaction {
+        val context = dfe.getContext<GraphQLContext>()
+        val jwtPayload = context.enforceSignedIn()
+        val admin =
+            AdministratorEntity.findById(jwtPayload.adminId)
+                ?: throw UnauthorizedException()
+        if (!Authorizer.mayViewFreinetAgencyInformationInRegion(admin, regionId)) {
+            throw ForbiddenException()
+        }
+        val projectEntity = ProjectEntity.find { Projects.project eq project }.firstOrNull()
+            ?: throw ProjectNotFoundException(project)
+
+        if (projectEntity.project != EAK_BAYERN_PROJECT) {
+            throw NotEakProjectException()
+        }
```

Comment:
🔧 The `if` block below can be pulled up into a `?.let {}` block before the Elvis operator, thus no variable needs to be declared.

Reply:
hm can you give me an example how to do that @andrew8er 
When i put the let after `firstOrNull()` i dunno how i can include both exceptions in the expression

Reply:
This should be equivalent, if I'm not mistaken:
```kt
        ProjectEntity.find { Projects.project eq project }.firstOrNull()
            ?.let {
                if (it.project != EAK_BAYERN_PROJECT) {
                    throw NotEakProjectException()
                }        
            }
            ?: throw ProjectNotFoundException(project)
```

Reply:
ah alright i thought in can be solved without the using the `if`condition. thx